src/main/java/gregtech/common/tileentities/boilers/GT_MetaTileEntity_Boiler_Bronze.java

```diff
@@ -179,129 +184,144 @@ protected int getCooldownInterval() {
-        } else if (
+        int fuelItemCombustionEnergy = TileEntityFurnace.getItemBurnTime(mInventory[2]) / 10;
+        if (fuelItemCombustionEnergy <= 0) return;
+        ValidationResult<ItemStack> combustionResult = tryCombustFuel(mInventory[2]);
+        if (combustionResult.getType() == ValidationType.VALID) {
+            this.mProcessingEnergy += fuelItemCombustionEnergy;
+            if (combustionResult.getResult() != null)
+                aBaseMetaTileEntity.addStackToSlot(3, combustionResult.getResult());
+            aBaseMetaTileEntity.decrStackSize(2, 1);
+        }
+    }
+
+    @NotNull
+    private static ValidationResult<ItemStack> tryCombustFuel(@NotNull ItemStack stack) {
+        ValidationType valid = ValidationType.VALID;
+        ItemStack ashes = null;
+        if (isItemSolidCarbonFuelItem(stack)) {
+            ashes = combustSolidCarbonFuelItem(stack);
+        } else if (isItemSolidCarbonFuelBlock(stack)) {
+            ashes = combustSolidCarbonFuelBlock(stack);
+        } else if (isDenseSolidFuel(stack)) {
+            ashes = combustDenseSolidFuel(stack);
+        } else {
+            valid = ValidationType.INVALID;
+        }
+        return ValidationResult.of(valid, ashes);
+    }
+
+    private static boolean isItemSolidCarbonFuelItem(@NotNull ItemStack fuelStack) {
+        return (GT_Utility.isPartOfMaterials(fuelStack, Materials.Coal)
+            && !GT_Utility.isPartOfOrePrefix(fuelStack, OrePrefixes.block))
+            || (GT_Utility.isPartOfMaterials(fuelStack, Materials.Charcoal)
+                && !GT_Utility.isPartOfOrePrefix(fuelStack, OrePrefixes.block))
+            || (GT_Utility.isPartOfMaterials(fuelStack, Materials.Lignite)
+                && !GT_Utility.isPartOfOrePrefix(fuelStack, OrePrefixes.block))
+            || (GT_Utility.isPartOfMaterials(fuelStack, Materials.Diamond)
+                && !GT_Utility.isPartOfOrePrefix(fuelStack, OrePrefixes.block))
+            || GT_OreDictUnificator.isItemStackInstanceOf(fuelStack, "fuelCoke")
+            || GT_OreDictUnificator.isItemStackInstanceOf(fuelStack, "fuelCactusCharcoal")
+            || GT_OreDictUnificator.isItemStackInstanceOf(fuelStack, "fuelCactusCoke")
+            || GT_OreDictUnificator.isItemStackInstanceOf(fuelStack, "fuelSugarCharcoal")
+            || GT_OreDictUnificator.isItemStackInstanceOf(fuelStack, "fuelSugarCoke");
+    }
+
+    private static ItemStack combustSolidCarbonFuelItem(@NotNull ItemStack stack) {
+        return XSTR.XSTR_INSTANCE.nextInt(
+            GT_Utility.isPartOfMaterials(stack, Materials.Coal)
+                || GT_Utility.isPartOfMaterials(stack, Materials.Charcoal) ? 3
+                    : GT_Utility.isPartOfMaterials(stack, Materials.Lignite) ? 8 : 2)
+            == 0 ? GT_OreDictUnificator.get(OrePrefixes.dustTiny, (GT_Utility.isPartOfMaterials(stack, Materials.Lignite) || GT_Utility.isPartOfMaterials(stack, Materials.Coal)) ? Materials.DarkAsh : Materials.Ash, 1L) : null;
```

Comment:
I suggest caching the results of `GT_Utility.isPartOfMaterials(...)` method. This will make this method much more readable.
```suggestion
        boolean isCoal = GT_Utility.isPartOfMaterials(stack, Materials.Coal);
        boolean isCharcoal = GT_Utility.isPartOfMaterials(stack, Materials.Charcoal);
        boolean isLignite = GT_Utility.isPartOfMaterials(stack, Materials.Lignite);
        return XSTR.XSTR_INSTANCE.nextInt(isCoal || isCharcoal ? 3 : isLignite ? 8 : 2) == 0
             ? GT_OreDictUnificator.get(OrePrefixes.dustTiny, (isCoal || isLignite) ? Materials.DarkAsh : Materials.Ash, 1L)
              : null;
```